### PR TITLE
Add WS-Trust test

### PR DESCRIPTION
**Pull request by ppalaga**
This ended up to be quite a big refactoring in the core module. I have split the processing for server side and clients. I hope the resulting design is a bit better to read and maintain. The main motivation was that the `@WebServiceProvider` annotated STS did not deploy properly and it was hard to follow in the code what is going on. A review would be welcome!

### CODE REVIEW

**Comment by shumonsharif** [comment]
Hey @ppalaga This is some excellent work; thanks for the effort to clean up the core module!  

I'll try to do a deeper dive later this weekend, but I'm seeing a lot of LGPL license headers ([such as this](https://github.com/quarkiverse/quarkus-cxf/pull/543/files#diff-555b01113e44b0172d8a357f76dde623784b6b80abb9ab6d4880245c9532f8cdR1-R21)) showing up in the repo.

I was under the impression that you can't copy LGPL licensed code into an Apache licensed project (though the other way is okay)?  Was just wondering given the headers on some of the newly introduced classes; is this worth cleaning up?

**Comment by ppalaga** [comment]
Let me merge like this because I have new changes on top of this. A review would still be appreciated.

**Comment by shumonsharif** [comment]
Hey @ppalaga I just reviewed a bit more thoroughly, excellent work, good clean code, and much easier to follow than previously ... it all looks good to me!  👍
